### DISCUSSION TITLE
Hide Pages for users not signed in

### DISCUSSION THREAD

**Discussion by JWHBerlin** [typemill/typemill#496] [Q&A] [answered]
I love typemill, since I found it a few weeks ago. A great system! Now I set up a small private page. I like to show some pages to the world, the most only for users, that are signed in. It works by standard, but I don‘t like that the world sees all page titles. Can I hide these titles/page names for not logged in users?
Best regards and thanks!

**Comment by trendschau** [comment, marked as answer]
Thank you, very glad to hear that! You mean to hide those pages in the navigation? I understand the requirement, unfortunately this is pretty complicated because the navigation is cached statically. To individualize the navigation items depending on the right and roles that are defined for each indivudal page I have to process the whole navigation on each call again. This will become very slow especially for larger websites. Anyway, I will add this to the list of features, but I doubt that I will find a quick solution for this, so it might take quite some time, sorry...

**Comment by trendschau** [comment]
While I am writing this: There is another feature, a checkbox to hide pages from the navigation generally (with and without authentication). It should be possible to enhance this feature so that hidden pages are still hidden for non-authenticated users, but visible for all authenticated users independent from the role of that user. The content will still be restricted to users without sufficient rights, but the navigation is visible for all authenticated users then. 

Not super sure if this behavior is confusing for admins or editors, but it might be a solution for the majority of requirements like yours.

**Reply by JWHBerlin** [reply]
Thanks for the ideas; I will try the 2nd proposition. And if it is not working, I can set up a small static Website as One-Pager before the closed area. Thanks!

**Reply by trendschau** [reply]
Ah, this does not work yet, hidden pages are hidden for all users right now, but it might be a comparatively easy solution, so I could try to add it to the release for the next month ....

**Reply by Bob-TheOwl** [reply]
This sounds like a good workaround. Is there anything new on this topic. I'm trying to archive the same as JWHBerlin.
Thanks. :-)

**Reply by trendschau** [reply]
Ahh, good hint, to be honest I forgot about this. I will now add it to my internal ticket and try to implement an option with the next release coming mid July...

**Reply by Bob-TheOwl** [reply]
Wow cool. Thanks. :-)
Typemill is really well-thought-out. Good job. Looking forward seeing, what is to come.

**Comment by trendschau** [comment]
In release 2.18.0 there is a new option in the access tab of the settings to hide pages from the navigation in frontend if the user has not enough rights. I also updated the documentation: https://docs.typemill.net/author-guide/restrict-pages

The new feature checks the role/rights of the current user and removes the page from the navigation on the fly if it has restrictions and if the user rights are not enough. Please note that this feature might slow down the performance of large websites since it has some complex extra checks.

Please try ...

**Reply by JWHBerlin** [reply]
Thanks a lot, that is great!

**Comment by Bob-TheOwl** [comment]
Awesome. That works really well.
Thank you. :-)
